Added support for waiting for other processes

exterrestris opened the pull request:
Detect lock files from other processes that are/may be modifying the data disks and wait for them to be removed before continuing with the diff/sync.

I use StableBit Drivepool to pool my data disks, and have a landing zone to help prevent writes to the data disks while a sync is underway (and for protection against disk failure). This is configured to 'rebalance' onto the main data disks daily before the scheduled sync, with a file created to indicate that a balance is in progress - using the functionality added in this PR, I can get the script to pause until the balance has completed. As another example, it could also be used to wait for a long-running backup to complete.

exterrestris committed:
Added support for waiting for other processes

Detect lock files from other processes that are/may be modifying the data disks and wait for them to be removed before continuing with the diff/sync

exterrestris committed:
Show which process is being waited for

exterrestris committed:
Fixed bug where last file was always waited for

Fixed bug where script would wait for last defined file even if it didn't exist

Chronial commented:
What is the advantage of having this functionality in snapraid-runner compared to having a wrapper-script?

exterrestris commented:
Primarily, the benefit is that it ties into the same logging and email functionality and configuration, which avoids the need to duplicate that functionality and configuration. It also makes it's clear as to what happens during a run - whether it is aborted completely, or merely delayed, together with with the cause.
Besides snapraid-runner is itself a wrapper script - creating yet another wrapper around it feels unnecessary.

Chronial commented:
Would #50 address your use-case?

exterrestris commented:
Apologies for taking so long to reply. Since creating this PR I've switched my home server from Windows to Linux, and changed the structure of my disks such that this is no longer really an issue for me. Having said that, #50 would still be a useful addition for me for other reasons - for this use case, as long as the 'pre-hook' could abort the sync then yes it would address it
